src/stan/callbacks/unique_stream_writer.hpp

```diff
@@ -64,9 +70,31 @@ class unique_stream_writer final : public writer {
    * Note: the precision of the output is determined by the settings
    *  of the stream on construction.
    *
-   * @param[in] state Values in a std::vector
+   * @param[in] values Values in a std::vector
+   */
+  void operator()(const std::vector<double>& values) {
+    if (output_ == nullptr)
+      return;
+    write_vector(values);
+  }
+
+  /**
+   * Writes multiple rows and columns of values in csv format.
+   *
+   * Note: the precision of the output is determined by the settings
+   *  of the stream on construction.
+   *
+   * @param[in] values A matrix of values. The input is expected to have
+   * parameters in the rows and samples in the columns. The matrix is then
+   * transposed for the output.
    */
-  void operator()(const std::vector<double>& state) { write_vector(state); }
+  void operator()(const Eigen::MatrixXd& values) {
+    if (output_ == nullptr)
+      return;
+    Eigen::IOFormat CommaInitFmt(Eigen::StreamPrecision, Eigen::DontAlignCols,
+                                 ", ", "", "", "\n", "", "");
+    *output_ << values.transpose().format(CommaInitFmt);
+  }
```

Comment:
Does this flatten the matrix into a single row of output or output the matrix as a 2-d object?

I know this PR is not primarily about the CSV writer, but this seems like an odd method to me to add, with the implicit transpose and the required contract that the number of rows is the same as the length of the header of the file.